### DISCUSSION TITLE
atom types for coarse graining

### DISCUSSION THREAD

**Discussion by KP-303** [mir-group/allegro#77] [Q&A]
If I want to use allegro to learn a coarse grained potential, how do I assign atom types in my dataset? Nequip insists on knowing the type of every atom. I could just lie and call them all carbon, but will this interfere with the training? Just checking there isn't something in the background that depends on atom type.

Thanks

**Comment by DavidW99** [comment]
No, it should not. The atom type is encoded by one-hot encoding. It gives a vector with the length of the total number of atom types and it is 1 at the specie's index and 0 elsewhere. You will assign the chemical symbol to the type index in the `yaml` file right after you specify your dataset.

```
# A mapping of chemical species to type indexes is necessary if the dataset is provided with atomic numbers instead of type indexes.
chemical_symbol_to_type:
  C: 0
```

Therefore, NequIP will not know your atomic type but only the type index that you assign.

**Comment by Linux-cpp-lisp** [comment]
Note also that you can just provide arbitrary type names, in order, instead of `chemical_symbol_to_type`:
```
type_names:
 - Type1
 - SomethingElse
```
where here type `0` will be named `Type1`, etc.

In this case, however, the input data must include an `atom_type` field instead of `atomic_numbers`.  In the extxyz format, this can be done with a custom per-atom `atom_type` field, and just giving `X` as the "species" (atomic number 0) for the atomic numbers of all atoms, for example.

I think the ZBL potential should be the only provided module that depends on the atomic number right now in a physical way.

**Reply by KP-303** [reply]
OK thanks.My beads can have a charge and still be the same bead type as the uncharged version. I have included a charge column in my extxyz training data. How do I make Nequip use the charge as a descriptor in the training? Or should I just relabel charged beads to be a different 'atom type'?

**Reply by Linux-cpp-lisp** [reply]
If you have a set of discrete particle types that just happen to be distinguished also by some notion of charge here, yes, relabeling them as different "atom types" would be reasonable.
